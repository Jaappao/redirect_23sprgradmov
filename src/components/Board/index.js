// @flow

import React from 'react'
import TimeRow from '../TimeRow'
import type { Period } from '../../types'
import config from '../../config'

import styled from 'styled-components'

const Wrapper = styled.div`
	padding: 10px;
	background: ${config.color.sub};
`
const Board = ({ periods }: { periods: Period[] }) => {
	return (
		<Wrapper>
			<p>開館 07:30</p>
			{periods.map(period => (
				<div>
					<TimeRow
						nextBreak={[2, 5].includes(period.info.period)}
						key={period.info.period}
						period={period}
					/>
				</div>
			))}
			<p>閉館 22:30</p>
		</Wrapper>
	)
}

export default Board
