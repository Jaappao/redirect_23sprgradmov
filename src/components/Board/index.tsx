import React from 'react'
import styled from 'styled-components'
import TimeRow from '../TimeRow'
import { Period } from '../../types'
import config from '../../config'

const Style = styled.div`
	padding: 10px;
	background: ${config.color.sub};

	display: flex;
	justify-content: center;
	> div {
		min-width: 300px;
		width: 100%;
		max-width: 600px;
		> p {
			font-weight: 800;
		}
	}
`

const Board = ({ periods }: { periods: Period[] }) => {
	return (
		<Style>
			<div>
				{periods.map((period, i) => (
					<TimeRow key={i} period={period} />
				))}
			</div>
		</Style>
	)
}

export default Board
