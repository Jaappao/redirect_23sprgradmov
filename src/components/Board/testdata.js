import moment from 'moment'

const periodsData = [
	{
		info: {
			period: '1',
			start: { h: 9, m: 20 },
			end: { h: 11, m: 0 },
		},
		status: { type: 'finish' },
		start: '2018-04-17T00:20:00.000Z',
		end: '2018-04-17T02:00:00.000Z',
	},
	{
		info: {
			period: '2',
			start: { h: 11, m: 10 },
			end: { h: 12, m: 50 },
		},
		status: { type: 'finish' },
		start: '2018-04-17T02:10:00.000Z',
		end: '2018-04-17T03:50:00.000Z',
	},
	{
		info: {
			period: '3',
			start: { h: 13, m: 40 },
			end: { h: 15, m: 20 },
		},
		status: { type: 'progress', progress: 75, rate: 0.75 },
		start: '2018-04-17T04:40:00.000Z',
		end: '2018-04-17T06:20:00.000Z',
	},
	{
		info: {
			period: '4',
			start: { h: 15, m: 30 },
			end: { h: 17, m: 10 },
		},
		status: { type: 'before', fromNowStr: '35分後' },
		start: '2018-04-17T06:30:00.000Z',
		end: '2018-04-17T08:10:00.000Z',
	},
	{
		info: {
			period: '5',
			start: { h: 17, m: 20 },
			end: { h: 19, m: 0 },
		},
		status: { type: 'before', fromNowStr: '2時間後' },
		start: '2018-04-17T08:20:00.000Z',
		end: '2018-04-17T10:00:00.000Z',
	},
	{
		info: {
			period: 'D5',
			start: { h: 18, m: 0 },
			end: { h: 19, m: 40 },
		},
		status: { type: 'before', fromNowStr: '3時間後' },
		start: '2018-04-17T09:00:00.000Z',
		end: '2018-04-17T10:40:00.000Z',
	},
	{
		info: {
			period: 'D6',
			start: { h: 19, m: 50 },
			end: { h: 21, m: 30 },
		},
		status: { type: 'before', fromNowStr: '5時間後' },
		start: '2018-04-17T10:50:00.000Z',
		end: '2018-04-17T12:30:00.000Z',
	},
]

const periods = periodsData.map(p => ({
	...p,
	start: moment(p.start),
	end: moment(p.end),
}))

export default periods
