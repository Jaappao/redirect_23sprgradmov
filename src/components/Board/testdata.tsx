import { Period } from '../../types'

const periodsData: Period[] = [
	{
		info: {
			period: '1',
			start: { h: 9, m: 20 },
			end: { h: 11, m: 0 },
		},
		status: { type: 'finish' },
	},
	{
		info: {
			period: '2',
			start: { h: 11, m: 10 },
			end: { h: 12, m: 50 },
		},
		status: { type: 'finish' },
	},
	{
		info: {
			period: '3',
			start: { h: 13, m: 40 },
			end: { h: 15, m: 20 },
		},
		status: { type: 'progress', progress: 75, rate: 0.75 },
	},
	{
		info: {
			period: '4',
			start: { h: 15, m: 30 },
			end: { h: 17, m: 10 },
		},
		status: { type: 'before' },
	},
	{
		info: {
			period: '5',
			start: { h: 17, m: 20 },
			end: { h: 19, m: 0 },
		},
		status: { type: 'before' },
	},
	{
		info: {
			period: 'D5',
			start: { h: 18, m: 0 },
			end: { h: 19, m: 40 },
		},
		status: { type: 'before' },
	},
	{
		info: {
			period: 'D6',
			start: { h: 19, m: 50 },
			end: { h: 21, m: 30 },
		},
		status: { type: 'before' },
	},
]

const periods = periodsData.map(p => ({
	...p,
}))

export default periods
