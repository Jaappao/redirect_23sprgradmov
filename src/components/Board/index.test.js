// @flow
import React from 'react'
import Board from '.'
import renderer from 'react-test-renderer'
import periods from './testdata'
import 'moment/locale/ja'

import 'jest-styled-components'

it('renders correctly', () => {
	const tree = renderer.create(<Board periods={periods} />).toJSON()

	expect(tree).toMatchSnapshot()
})
