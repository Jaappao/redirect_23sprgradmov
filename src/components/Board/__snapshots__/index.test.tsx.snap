// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c10 {
  width: 100%;
  height: 5px;
  background: #004645;
  border-radius: 5px;
}

.c11 {
  width: 100%;
  height: 100%;
  background: #a6c0bf;
  border-radius: 5px;
}

.c13 {
  width: 75%;
  height: 100%;
  background: #a6c0bf;
  border-radius: 5px;
}

.c15 {
  width: 0%;
  height: 100%;
  background: #a6c0bf;
  border-radius: 5px;
}

.c2 {
  margin-top: 10px;
}

.c2[data-nextspace='true'] {
  border-bottom: #717b8a dashed;
  padding-bottom: 5px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5em;
  font-weight: 700;
  padding: 2.5px 0;
}

.c5 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c6 {
  font-size: 0.6em;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  padding-left: 5px;
}

.c7 {
  font-size: 0.6em;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c8 {
  margin-left: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  width: 50px;
  margin-left: 5px;
  color: #a6ccff;
}

.c9 {
  color: gray;
}

.c12 {
  color: #ebe971;
}

.c14 {
  color: black;
}

.c0 {
  padding: 10px;
  background: #0a1325;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  min-width: 300px;
  width: 100%;
  max-width: 600px;
}

.c1 > p {
  font-weight: 800;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p>
      開館 08:00
    </p>
    <div
      className="c2"
      data-nextspace={false}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          1
          .
        </div>
        <div
          className="c5"
        >
          09:20
           - 
          11:00
        </div>
        <div
          className="c6"
        />
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="c9"
              color="gray"
            >
              Fin
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <div
          className="c11"
        />
      </div>
    </div>
    <div
      className="c2"
      data-nextspace={true}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          2
          .
        </div>
        <div
          className="c5"
        >
          11:10
           - 
          12:50
        </div>
        <div
          className="c6"
        />
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="c9"
              color="gray"
            >
              Fin
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <div
          className="c11"
        />
      </div>
    </div>
    <div
      className="c2"
      data-nextspace={false}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          3
          .
        </div>
        <div
          className="c5"
        >
          13:40
           - 
          15:20
        </div>
        <div
          className="c6"
        >
          <span>
            (
            25
            min)
          </span>
        </div>
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="c12"
              color="#ebe971"
            >
              Now
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <div
          className="c13"
        />
      </div>
    </div>
    <div
      className="c2"
      data-nextspace={false}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          4
          .
        </div>
        <div
          className="c5"
        >
          15:30
           - 
          17:10
        </div>
        <div
          className="c6"
        />
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="c14"
              color="black"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <div
          className="c15"
        />
      </div>
    </div>
    <div
      className="c2"
      data-nextspace={true}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          5
          .
        </div>
        <div
          className="c5"
        >
          17:20
           - 
          19:00
        </div>
        <div
          className="c6"
        />
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="c14"
              color="black"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <div
          className="c15"
        />
      </div>
    </div>
    <div
      className="c2"
      data-nextspace={false}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          d5
          .
        </div>
        <div
          className="c5"
        >
          18:00
           - 
          19:40
        </div>
        <div
          className="c6"
        />
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="c14"
              color="black"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <div
          className="c15"
        />
      </div>
    </div>
    <div
      className="c2"
      data-nextspace={false}
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          d6
          .
        </div>
        <div
          className="c5"
        >
          19:50
           - 
          21:30
        </div>
        <div
          className="c6"
        />
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="c14"
              color="black"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <div
          className="c15"
        />
      </div>
    </div>
    <p>
      閉館 22:30
    </p>
  </div>
</div>
`;
